extensions/cache/src/main/java/org/radargun/stages/cache/background/BackgroundStatisticsManager.java

```diff
@@ -82,8 +83,19 @@ public synchronized void startStats() {
 
    public synchronized void stopStats() {
       if (statsTask != null) {
-         statsTask.cancel(true);
-         statsTask = null;
+
+         if (statsTask.cancel(true)) {
+            statsTask = null;
+         } else {
+            log.error("Statistics task was not cancelled");
+         }
+         Utils.shutdownAndWait(executor);
```

Comment:
If you terminate the executor, please make sure that a new one is created in `startStats`. The stats can be started/stopped several times during a scenario.

Reply:
@alanfx Then, what does `startStats()` do when `executor` is null?

Reply:
@rvansa I see what you're saying now, sorry. Now `BackgroundStatisticsManager.stopStats()` does not set `executor` to `null`, so it is still available to schedule tasks in `BackgroundStatisticsManager.startStats()`.

Reply:
@alanfx I am afraid that since you shutdown the executor here, task scheduled in `startStats` will be rejected, though. 

Maybe a test case would get this correct?

Reply:
@rvansa fair point, I'll see what I can find out

Reply:
@rvansa The [JavaDocs](https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ExecutorService.html) for `ExecutorService` are pretty clear that shutting down prevents new tasks from being submitted, so I updated the code to re-initialize `executor` in `BackgroundStatisticsManager.stopStats()`. Thanks for catching this.